app/services/badge_granter.rb

```diff
@@ -12,6 +12,40 @@ def self.grant(badge, user, opts = {})
     BadgeGranter.new(badge, user, opts).grant
   end
 
+  def self.mass_grant(badge, users)
+    return unless badge.enabled?
+
+    system_user_id = Discourse.system_user.id
+    user_badges = users.map { |u| { badge_id: badge.id, user_id: u.id, granted_by_id: system_user_id, granted_at: Time.now } }
+    granted_badges = UserBadge.insert_all(user_badges, returning: %i[user_id])
+
+    users_to_notify = users.each do |user|
+      use_default_locale = !SiteSetting.allow_user_locale && user.locale.blank?
+      notification_locale = use_default_locale ? SiteSetting.default_locale : user.locale
+
+      notification = I18n.with_locale(notification_locale) do
```

Comment:
This feels a little repetitive from the #grant method below - could you split it out to a function and use it in both places?